agents/craftassist/tests/interactive_memory_explorer.py

```diff
@@ -33,12 +33,9 @@ def add_many_objects(test):
 class MemoryExplorer(BaseCraftassistTestCase):
     def setUp(self):
         super().setUp()
-        dummy_dialogue_stack = DialogueStack(self.agent, self.agent.memory)
+        dummy_dialogue_stack = DialogueStack(self.agent.memory)
```

Comment:
looks like memory has an attribute called : `dialogue_stack` which is an object of the class `DialogueStack` (defined in a file : `dialogue_stack`) and `DialogueStack` takes memory as an arg. This can get confusing pretty quick. Is there a way to disambiguate these a bit with the naming perhaps ?
Also, can the `DialogueStack` taking memory as an arg be worked around? Only the `add_new` method there really needs access to memory. We can perhaps move that out to agent or DialogueManager ?

Reply:
fixed in the stingiest way in c576a26cca76c32b5ce4fdfd9d89faec68c9d96b.  all this will be redone when DialogueObjects become Tasks in a few weeks...